### DISCUSSION TITLE
Suggestion: CamelCase tags to improve experience for screen readers

### DISCUSSION THREAD

**Discussion by lpar** [mastodon/mastodon#19692] [General]
I recently became aware that screen reader software has a much easier time reading hashtags if the words in the tag are CamelCased.

I notice that when I try to enter a hashtag camelcased, Mastodon actively discourages me by showing me a downcased version. I'd like to suggest that at a minimum, it shouldn't do this.

And in fact, I think ideally Mastodon would have dictionaries and suggest CamelCased versions of hashtags algorithmically. It wouldn't always be right, but it would normalize and promote the idea and make life on the Fediverse just a bit more pleasant for people using screen readers.

**Comment by Gargron** [comment]
> I notice that when I try to enter a hashtag camelcased, Mastodon actively discourages me by showing me a downcased version. I'd like to suggest that at a minimum, it shouldn't do this.

Mastodon saves hashtags in the way they were first used. What it is showing you is an autocompletion based on that. You could be typing all lowercase and it would suggest you the camelCase version if that's what the hashtag is saved as. After you've used the tag yourself, the page also saves your choices in local storage, so next time the autocompletion should come up with the way you typed it.

Admins can technically fix casing of individual hashtags, but the UI to get to a particular hashtag, unless it is trending, is not there right now. I acknowledge it would be ideal if hashtags could self-correct based on current usage somehow (but that seems prone to abuse as well, e.g. poisoning the database with `eXaMpLe`).

**Comment by lpar** [comment]
That was why I was thinking of using a dictionary to calculate likely camelcased versions. For example, for the word "camelcased" it would find "camel" in the dictionary and switch it to "Camel", then find "case" and switch it to "Case". There would obviously be tags it would fail on, like "thepenismightierthanthesword", but we're only talking autocompletion suggestions here, rather than imposed capitalization.

**Comment by jocelynetienne** [comment]
To answer @Gargron's caveats, I believe keywords already present in both downcase and camelcase versions could:
a/ have their frequency summed when it comes to calculating their popularity,
b/ if the camelcase version counts for, say, at least 10% of the total frequency (or some 100 recent occurences, whichever is less), then the version suggested should be the camelcase one.

**Comment by SmallBlueElephant** [comment]
There is a corresponding Issue https://github.com/mastodon/mastodon/issues/20267. 
Please have a look there and take part in the discussion.
